### PR TITLE
Inspector: Download logs for manual processing

### LINKED ISSUE #28752: Download Logs for manual processing
**What would you like to be added**:

From Grafana Dashboard Logs Panel or from Explore Logs, I would like to export the logs as a plain file to process/analyze/archive/forward logs manually.

**Why is this needed**:

The use case is that we would like to filter with context. you can easily process text logs with` grep bla -C 5` or `| grep foo -B 2 -A 20`. The log panel for grafana dashboards very limited features and also the Explore feature is limited to context on click on a single line.

Related Issue: https://github.com/grafana/loki/issues/1940

**Workaround**:

With logcli from Loki, the logs can be retrieved in usable format for further manual processing. But that requires configuring logcli, authenticating and manually copying the query and setting the range and limits again.

Example logcli query and output
`$ logcli query --since=5m '{category="dockerlogs",compose_project=~"webproxy",compose_service=~".*",host=~"dev2|test1"} |~ ""'`

```
https://loki.example.com/loki/api/v1/query_range?direction=BACKWARD&end=1604325539739801801&limit=30&query=%7Bcategory%3D%22dockerlogs%22%2Ccompose_project%3D~%22webproxy%22%2Ccompose_service%3D~%22.%2A%22%2Chost%3D~%22dev2%7Ctest1%22%7D+%7C~+%22%22&start=1604325239739801801
Common labels: {category="dockerlogs", compose_project="webproxy", compose_service="nginx-web", container_name="webproxy_nginx-web", source="stdout"}
2020-11-02T14:58:49+01:00 {filename="/var/log/docker/0a1fb7c3215e108da67f2f0c0299494506532466fd3d0a38f06c06423d6f0012/json.log", host="test1"} bar.test.example.com 213.102.102.58 - - [02/Nov/2020:13:58:49 +0000] "POST /dongle1/logbook/storage HTTP/1.1" 200 5 "-" "python-requests/2.12.4"
2020-11-02T14:58:47+01:00 {filename="/var/log/docker/1b845dad76bddc46b2a6f59772a5431825d591ad80865686d29d6bdd31f508c9/json.log", host="dev2"}  bar.dev.example.com 54.36.102.164 - - [02/Nov/2020:13:58:47 +0000] "POST /munic HTTP/1.1" 404 314 "-" "Go-http-client/1.1"
2020-11-02T14:58:45+01:00 {filename="/var/log/docker/0a1fb7c3215e108da67f2f0c0299494506532466fd3d0a38f06c06423d6f0012/json.log", host="test1"} foo.test1.example.com 88.198.147.151 - asdf-foo [02/Nov/2020:13:58:45 +0000] "GET /api/v1/allmetrics?format=prometheus HTTP/1.1" 200 542546 "-" "Prometheus/2.21.0"
2020-11-02T14:58:45+01:00 {filename="/var/log/docker/1b845dad76bddc46b2a6f59772a5431825d591ad80865686d29d6bdd31f508c9/json.log", host="dev2"}  foo.dev2.example.com 88.198.147.151 - asdf-foo [02/Nov/2020:13:58:45 +0000] "GET /api/v1/allmetrics?format=prometheus HTTP/1.1" 200 507076 "-" "Prometheus/2.21.0"
2020-11-02T14:58:38+01:00 {filename="/var/log/docker/0a1fb7c3215e108da67f2f0c0299494506532466fd3d0a38f06c06423d6f0012/json.log", host="test1"} bar.test.example.com 213.102.102.58 - - [02/Nov/2020:13:58:38 +0000] "POST /dongle1/logbook/storage HTTP/1.1" 200 5 "-" "python-requests/2.12.4"
2020-11-02T14:58:30+01:00 {filename="/var/log/docker/0a1fb7c3215e108da67f2f0c0299494506532466fd3d0a38f06c06423d6f0012/json.log", host="test1"} foo.test1.example.com 88.198.147.151 - asdf-foo [02/Nov/2020:13:58:30 +0000] "GET /api/v1/allmetrics?format=prometheus HTTP/1.1" 200 542537 "-" "Prometheus/2.21.0"
2020-11-02T14:58:30+01:00 {filename="/var/log/docker/1b845dad76bddc46b2a6f59772a5431825d591ad80865686d29d6bdd31f508c9/json.log", host="dev2"}  foo.dev2.example.com 88.198.147.151 - asdf-foo [02/Nov/2020:13:58:30 +0000] "GET /api/v1/allmetrics?format=prometheus HTTP/1.1" 200 507047 "-" "Prometheus/2.21.0"
2020-11-02T14:58:27+01:00 {filename="/var/log/docker/0a1fb7c3215e108da67f2f0c0299494506532466fd3d0a38f06c06423d6f0012/json.log", host="test1"} bar.test.example.com 213.102.102.58 - - [02/Nov/2020:13:58:27 +0000] "POST /dongle1/logbook/storage HTTP/1.1" 200 5 "-" "python-requests/2.12.4"
2020-11-02T14:58:15+01:00 {filename="/var/log/docker/0a1fb7c3215e108da67f2f0c0299494506532466fd3d0a38f06c06423d6f0012/json.log", host="test1"} foo.test1.example.com 88.198.147.151 - asdf-foo [02/Nov/2020:13:58:15 +0000] "GET /api/v1/allmetrics?format=prometheus HTTP/1.1" 200 542541 "-" "Prometheus/2.21.0"
2020-11-02T14:58:15+01:00 {filename="/var/log/docker/0a1fb7c3215e108da67f2f0c0299494506532466fd3d0a38f06c06423d6f0012/json.log", host="test1"} bar.test.example.com 213.102.102.58 - - [02/Nov/2020:13:58:15 +0000] "POST /dongle1/logbook/storage HTTP/1.1" 200 5 "-" "python-requests/2.12.4"

```

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
This PR adds **Download logs for manual processing** functionality. We show **Download logs** button only if we have dataFrames with logs. 

**Explore: Data tab when logs data (button visible):**
image

**Explore: Data tab without logs data (button not visible):**
image

**Panel: Data tab when logs data (button visible):**
image

**Panel: Data tab without logs data (button not visible):**
image

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/28752

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/inspector/InspectDataOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
+import React, { FC } from 'react';
```
Comment (ivanahuckova) [inline public/app/features/inspector/InspectDataOptions.tsx:1]: This is just moved from `public/app/features/inspector/InspectDataTab.tsx`, so **InspectDataTab** component is smaller.

File: public/app/features/inspector/InspectDataTab.tsx [hunk 2]
```diff
@@ -128,129 +151,9 @@ export class InspectDataTab extends PureComponent<Props, State> {
     });
   }
 
-  getActiveString() {
```
Comment (ivanahuckova) [inline public/app/features/inspector/InspectDataTab.tsx:131]: Moved to `public/app/features/inspector/InspectDataOptions.tsx`

File: public/app/features/inspector/InspectDataTab.tsx [hunk 3]
```diff
@@ -99,6 +98,24 @@ export class InspectDataTab extends PureComponent<Props, State> {
     saveAs(blob, fileName);
   };
 
+  exportLogsAsTxt = () => {
```
Comment (ivanahuckova) [inline public/app/features/inspector/InspectDataTab.tsx:101]: This is a new thing that this PR adds - `exportLogsAsTxt`.

File: public/app/features/inspector/InspectDataTab.tsx [hunk 4]
```diff
@@ -269,12 +172,25 @@ export class InspectDataTab extends PureComponent<Props, State> {
 
     // let's make sure we don't try to render a frame that doesn't exists
     const index = !dataFrames[dataFrameIndex] ? 0 : dataFrameIndex;
-    const data = dataFrames[index];
+    const dataFrame = dataFrames[index];
+    const hasLogs = dataFrames.some((df) => df?.meta?.preferredVisualisationType === 'logs');
```
Comment (ivanahuckova) [inline public/app/features/inspector/InspectDataTab.tsx:176]: If any of dataFrames is logs, show `Download logs` button.

**Review by hugohaggmark — APPROVED** [review]
Looks good 🎉 !

**Comment by ivanahuckova** [comment]
I have created separate issue for the problem with headerElement overflowing the row and hiding title https://github.com/grafana/grafana/issues/32893

**Comment by achatterjee-grafana** [comment]
@ivanahuckova, will you open a doc PR to capture the changes in this PR? Thanks!

**Comment by 123BLiN** [comment]
Sorry for silly question, will this allow to download all logs for specified period no matter of the size, or only last 5k lines?

**Comment by ivanahuckova** [comment]
Only received logs respecting the limit.

**Comment by Azbesciak** [comment]
> Only received logs respecting the limit.

Maybe some way to download logs directly via API, but without limits? I just want to do the same like entering the direct log location and copying them. All available log.

**Pull request merged**